src/main/java/org/primeframework/mvc/sampleapp/action/BaseAction.java

```diff
@@ -1,8 +1,12 @@
 package org.primeframework.mvc.sampleapp.action;
 
+import org.primeframework.mvc.action.result.annotation.Forward;
 import org.primeframework.mvc.action.result.annotation.SaveRequest;
 
 // by default, this saves request and redirects the user to / if they are unauthenticated
-@SaveRequest(uri = "/")
+@SaveRequest(uri = "/login")
+@Forward.List(
+        @Forward(code = "error", page = "error.ftl", status = 500)
+)
```

Comment:
Can you run the formatter on this file, or if nothing moves after formatting, export the project formatter from another project, and import it here.  

Once you confirm the formatter is in place, maybe just run it on the entire project, looks like some other files need formatting as well. You can click on directories in IJ and format. 

We generally exclude `*.ftl` from any formatting rules because IJ seems to suck at formatting FreeMarker files, and it usually just breaks them.

Reply:
Done